mobile/src/main/java/rs/readahead/washington/mobile/views/activity/VideoViewerActivity.kt

```diff
@@ -0,0 +1,540 @@
+        super.onStart()
+        if (SDK_INT > 23) {
+            initializePlayer()
+        }
+    }
+
+    public override fun onResume() {
+        super.onResume()
+        if (SDK_INT <= 23 || player == null) {
+            initializePlayer()
+        }
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        if (SDK_INT <= 23) {
+            releasePlayer()
+        }
+    }
+
+    public override fun onStop() {
+        super.onStop()
+        if (SDK_INT > 23) {
+            releasePlayer()
+        }
+    }
+
+    override fun onDestroy() {
+        if (alertDialog != null) {
+            if (alertDialog!!.isShowing) {
+                alertDialog!!.dismiss()
+            }
+        }
+        hideProgressDialog()
+        if (presenter != null) {
+            presenter!!.destroy()
+        }
+        super.onDestroy()
+    }
+
+    @NeedsPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+    fun exportMediaFile() {
+        if (vaultFile != null && presenter != null) {
+            if (vaultFile?.metadata != null) {
+                showExportWithMetadataDialog()
+            } else {
+                withMetadata = false
+                maybeChangeTemporaryTimeout {
+                    performFileSearch()
+                }
+            }
+        }
+    }
+
+    private fun performFileSearch() {
+        if (hasStoragePermissions(this)) {
+            if (Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP) {
+                val intent = Intent(Intent.ACTION_OPEN_DOCUMENT_TREE)
+                intent.addFlags(
+                    Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION or Intent.FLAG_GRANT_WRITE_URI_PERMISSION or Intent.FLAG_GRANT_READ_URI_PERMISSION
+                )
+                startActivityForResult(intent, PICKER_FILE_REQUEST_CODE)
+            } else {
+                if (presenter != null) {
+                    presenter!!.exportNewMediaFile(withMetadata, vaultFile, null)
+                }
+            }
+        } else {
+            requestStoragePermissions()
+        }
+    }
+
+    private fun hasStoragePermissions(context: Context): Boolean {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            Environment.isExternalStorageManager()
+        } else {
+            val result: Int = ContextCompat.checkSelfPermission(
+                context, Manifest.permission.READ_EXTERNAL_STORAGE
+            )
+            val result1: Int = ContextCompat.checkSelfPermission(
+                context, Manifest.permission.WRITE_EXTERNAL_STORAGE
+            )
+            result == PackageManager.PERMISSION_GRANTED && result1 == PackageManager.PERMISSION_GRANTED
+        }
+    }
+
+    override fun onMediaExported() {
+        showToast(resources.getQuantityString(R.plurals.gallery_toast_files_exported, 1, 1))
+    }
+
+    override fun onExportError(error: Throwable) {
+        showToast(R.string.gallery_toast_fail_exporting_to_device)
+    }
```

Comment:
All toast should be replaced with snackbars, you can find it in `DialogUtils` in the shared ui module.